neoscore/western/chordrest.py

```diff
@@ -333,6 +333,38 @@ def extra_attachment_point(self) -> Point:
         x = notehead.x + bounding_rect.x + (bounding_rect.width / 2)
         return Point(x, y)
 
+    def mid_stem_attachment_point(self) -> Point:
+        """A mid-point where common attachments like tremolo's could go.
+
+        For chords, this is a point centered above or below the outermost notehead
+        opposite of the stem direction.
+
+        For rests, this is a point centered above the rest.
+
+        The returned point is relative to the Chordrest.
+        """
+        if self.rest:
+            bounding_rect = self.rest.bounding_rect
+            x = self.rest.x + bounding_rect.x + (bounding_rect.width / 2)
+            y = self.rest.y + bounding_rect.y - self.staff.unit(1)
+            return Point(x, y)
+        if self.stem_direction == DirectionY.DOWN:
+            notehead = self.lowest_notehead
+            bounding_rect = notehead.bounding_rect
+            x = notehead.x - bounding_rect.x
+            y = (
+                notehead.y
+                + bounding_rect.y
+                + bounding_rect.height
+                + self.staff.unit(1.5)
+            )
```

Comment:
this uses a fixed offset from the notehead column, but the method is called "mid-stem attachment point". this is not actually the middle of the stem. maybe it should be called "tremolo attachment point" instead? that naming would also make better sense for rests and stemless durations.

Reply:
it's not really clear anyway what the "middle of the stem" means, since the stem overlaps with the notehead column, flags, and beams, and different people may have different ideas of which of those areas are accounted for in this method.